src/uu/ls/src/ls.rs

```diff
@@ -3125,8 +3119,8 @@ fn display_item_name(
     }
 
     if config.format == Format::Long
-        && path.file_type(&mut state.out).is_some()
-        && path.file_type(&mut state.out).unwrap().is_symlink()
+        && path.file_type().is_some()
```

Comment:
Could use `is_some_and` here.

Reply:
Thanks.  Good solution for some code ugliness.